File: Linked List/83.Remove Duplicates from Sorted List/Solution.java
/**
 * 将已排序链表中值重复的结点只留其一，其余去除。
 * 题目地址：https://leetcode.com/problems/remove-duplicates-from-sorted-list/
**/
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null || head.next == null) return head;
        ListNode p = head, q = p.next;
        while(q != null) {
            if( p.val == q.val) {  //若重复则跳到下一个
                q = q.next;
            } else {  //不重复时，则去重并跳到下一个。
                p.next = q;
                p = q;
                q = p.next;
            }
        }
        p.next = null;
        return head;
    }
}

File: Linked List/83.Remove Duplicates from Sorted List/Solution.py
class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def deleteDuplicates(self, head):
        if not head or not head.next: 
            return head
        p = head
        temp = head.next
        while temp:
            if temp.val == p.val:
                temp = temp.next
            else:
                p.next = temp
                p = temp
                temp = p.next
        p.next = None
        return head